static/js/lib/api.js

```diff
@@ -6,6 +6,92 @@ import R from "ramda"
 
 import { S, parseJSON, filterE } from "./sanctuary"
 
+export const booksRequest = (force = false) => ({
+  url: "/api/library/books/",
+  update: {
+    books: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    // The response is a list of serialized books. This turns it into an object mapping
+    // the book id to the full serialized object: {[book.id]: [book]}
+    return {
+      books: R.fromPairs(
+        responseJson.map((book) => (
+          [book.id, book]
+        ))
+      )
+    }
+  },
+  force
+})
+
+export const deleteBookRequest = (bookId) => ({
+  url: `/api/library/books/${bookId}/`,
+  update:  {
+    books: (prev, next) => {
+      return R.omit(
+        [next.toDelete],
+        prev
+      )
+    },
+  },
+  transform: () => {
+    return {
+      books: {
+        toDelete: String(bookId)
+      }
+    }
+  },
+  options: {
+    method: "DELETE"
+  }
+})
```

Comment:
This demonstrates some fancier, not-exactly-standard usage of `update`/`transform` that saves us a request. When a book is successfully deleted, we don't really need to make another request to the books endpoint for the updated list. We can just remove the book from those results. This is done by using a custom key in `transform` to indicate the book id that needs to be removed, then using that key in `update` to omit it from the existing `books` state